### PR TITLE
Lenient SealedTx decoding

### PR DESCRIPTION
**Pull request by ghost**
> This is my very first PR in cardano-wallet so my apologies if I broke some rules of the land :)

This simple PR allows clients to provide either a base16 or base64 encoding when submitting a transaction for signing and other endpoints relevant for the new signing process. It is based off of @KtorZ's #2943 PR as this strand of work was triggered by us tripping ourselves in the foot trying to sign a base16 encoded transaction.

### CODE REVIEW

**Comment by ghost** [comment]
I guess the PR fails because of the commits it is based on? Will try to rebase on master

File: lib/core/test/unit/Cardano/Wallet/Api/TypesSpec.hs [hunk 1]
```diff
@@ -559,6 +561,23 @@ spec = parallel $ do
...
+            forAll selectFromPreparedBinaries $ \ bs ->
+                 let result = parseJSONSealedTx $ Aeson.encode $ ApiBytesT @'Base16 bs
+                  in result == Right bs &
+                     counterexample ("Parse result: " <> show result)
+
+        it "can decode from base-64 encoded string" $
+            forAll selectFromPreparedBinaries $ \ bs ->
+                 let result = parseJSONSealedTx $ Aeson.encode $ ApiBytesT @'Base64 bs
+                  in result == Right bs &
+                     counterexample ("Parse result: " <> show result)
```
Comment (KtorZ) [inline lib/core/test/unit/Cardano/Wallet/Api/TypesSpec.hs:579]: :+1: convincing properties.

File: lib/core/src/Cardano/Wallet/Api/Types.hs [hunk 2]
```diff
@@ -2671,7 +2671,9 @@ instance (HasBase base, ByteArrayAccess bs) => ToJSON (ApiBytesT base bs) where
     toJSON = String . toText @(ApiBytesT base bs)
 
 instance FromJSON (ApiT SealedTx) where
-    parseJSON = fmap ApiT . parseSealedTxBytes @'Base64
+    parseJSON v = do
+        tx <- parseSealedTxBytes @'Base16 v <|> parseSealedTxBytes @'Base64 v
+        pure $ ApiT tx
```
Comment (KtorZ) [inline lib/core/src/Cardano/Wallet/Api/Types.hs:2676]: :+1:

**Review by KtorZ — APPROVED** [review]

**Comment by ghost** [comment]
I did not run _all_ unit tests before pushing and some of them are failing. Will revert to draft status, apologies for the inconvenience.

**Review by paweljakubas — APPROVED** [review]
lgtm. The properties will be even better when `selectFromPreparedBinaries` is replaced with `genTx` from cardano-api:gen soon (@sevanspowell is working on this)

**Comment by ghost** [comment]
Need to fix a unit test, will push soon...

**Comment by ghost** [comment]
Sorry for the confusion, PR now complete with test fixed and documentation added in swagger.yaml.

**Pull request merged**